packages/run-protocol/src/vaultFactory/vaultManager.js

```diff
@@ -227,164 +312,169 @@ export const makeVaultManager = (
-    ));
-
-    /** @type {AssetState} */
-    const payload = harden({
-      compoundedInterest,
-      interestRate,
-      latestInterestUpdate,
-      totalDebt,
-    });
-    assetUpdater.updateState(payload);
+  liquidateAndRemove: ({ state }, [key, vault]) => {
+    const { debtMint, factoryPowers, penaltyPoolSeat, prioritizedVaults, zcf } =
+      state;
+    trace('liquidating', vault.getVaultSeat().getProposal());
+    state.liquidationInProgress = true;
 
-    trace('chargeAllVaults complete', payload);
+    // Start liquidation (vaultState: LIQUIDATING)
+    return liquidate(
+      zcf,
+      vault,
+      debtMint.burnLosses,
+      state.liquidationStrategy,
+      state.collateralBrand,
+      penaltyPoolSeat,
+      factoryPowers.getGovernedParams().getLiquidationPenalty(),
+    )
+      .then(() => {
+        prioritizedVaults?.removeVault(key);
+        state.liquidationInProgress = false;
+      })
+      .catch(e => {
+        state.liquidationInProgress = false;
+        // XXX should notify interested parties
+        console.error('liquidateAndRemove failed with', e);
+      });
+  },
+};
 
-    reschedulePriceCheck();
-  };
+const managerBehavior = {
+  /** @param {MethodContext} context */
+  getGovernedParams: ({ state }) => state.factoryPowers.getGovernedParams(),
 
   /**
-   * @param {Amount<'nat'>} toMint
-   * @throws if minting would exceed total debt
+   * @param {MethodContext} context
+   * @param {Amount<'nat'>} collateralAmount
    */
-  const checkDebtLimit = toMint => {
-    const debtPost = AmountMath.add(totalDebt, toMint);
-    const limit = loanParamGetters.getDebtLimit();
-    if (AmountMath.isGTE(debtPost, limit)) {
-      assert.fail(X`Minting would exceed total debt limit ${q(limit)}`);
-    }
-  };
-
-  const maxDebtFor = async collateralAmount => {
+  maxDebtFor: async ({ state }, collateralAmount) => {
+    const { debtBrand, priceAuthority } = state;
     const quoteAmount = await E(priceAuthority).quoteGiven(
       collateralAmount,
       debtBrand,
     );
     // floorDivide because we want the debt ceiling lower
     return floorDivideBy(
       getAmountOut(quoteAmount),
-      shared.getLiquidationMargin(),
+      state.factoryPowers.getGovernedParams().getLiquidationMargin(),
     );
-  };
-
+  },
+  /**
+   * TODO utility method to turn a callback into non-actual one
+   * was type {MintAndReallocate}
+   *
+   * @param {MethodContext} context
+   * @param {Amount} toMint
+   * @param {Amount} fee
+   * @param {ZCFSeat} seat
+   * @param {...ZCFSeat} otherSeats
+   * @returns {void}
+   */
+  mintAndReallocate: (
+    { state, facets: { helper } },
+    toMint,
+    fee,
+    seat,
+    ...otherSeats
+  ) => {
+    helper.checkDebtLimit(toMint);
+    state.factoryPowers.mintAndReallocate(toMint, fee, seat, ...otherSeats);
+    state.totalDebt = AmountMath.add(state.totalDebt, toMint);
+  },
   /**
+   * @param {MethodContext} context
+   * @param {*} toBurn
+   * @param {*} seat
```

Comment:
Why leave these as `*`?

```suggestion
   * @param {Amount} toBurn
   * @param {ZCFSeat} seat
```

Reply:
👍